pdns/dnsdist.hh

```diff
@@ -51,6 +59,123 @@ extern uint16_t g_ECSSourcePrefixV4;
 extern uint16_t g_ECSSourcePrefixV6;
 extern bool g_ECSOverride;
 
+
+
+// ----------------------------------------------------------------------------
+// GCA - Seth Ornstein - 5/30/2017 - new extra class in DNSQuestion struct
+
+class QTag
+{
+
+public:
+
+// ----------------------------------------------------------------------------
+// constructor
+// ----------------------------------------------------------------------------
+QTag()
+{
+}
+
+// ----------------------------------------------------------------------------
+// destructor - verify for debugging that it is called as this is how tags become freed
+// ----------------------------------------------------------------------------
+~QTag()
+{
+}
+
+// ----------------------------------------------------------------------------
+// add() - add a label and value to the tags
+// ----------------------------------------------------------------------------
+bool add(std::string strLabel, std::string strValue)
+{
+bool bStatus = true;
+
+   tagData.insert( {strLabel, strValue});
+//   tagData[strLabel] = strValue;
+   return(bStatus);
+}
+
+// ----------------------------------------------------------------------------
+// getMatch() - return the matching tag value
+// ----------------------------------------------------------------------------
+std::string getMatch(const std::string& strLabel)  const
+{
+
+    std::unordered_map<std::string, std::string>::const_iterator got =tagData.find (strLabel);
+    if(got == tagData.end())
+      {
+       return("");
+      }
+    else
+      {
+       return(got->second);
+      }
+}
+
+
+// ----------------------------------------------------------------------------
+// getEntry() - return the specified tag entry
+// ----------------------------------------------------------------------------
+std::string getEntry(int iEntry) const
```

Comment:
Please don't use a signed type for an unsigned value, `size_t` or at least `unsigned int` would be more appropriate.